src/client/components/organisms/BookingForm.tsx

```diff
@@ -19,14 +21,26 @@ export const BookingForm: React.FC = () => {
       location: Hotel.location,
     },
   });
-  const { isLoading, createBookingAsync } = useBookingStore();
+  const { changeModalStatus, modal } = useAppStore(); // Get modal state
+  const { isLoading, createBookingAsync, bookingCreateRes } = useBookingStore();
   const deviceType = useAppStore((state) => state.deviceType);
 
   const onSubmit = async (data: BookingDto) => {
     console.log(data);
     await createBookingAsync(data);
   };
 
+  // Watch for changes in bookingCreateRes
+  useEffect(() => {
```

Comment:
This can be simplified as:

```
useEffect(() => {
  if (bookingCreateRes?.url) {
    deviceType === 'desktop' ? changeModalStatus(ModalStatus.OPEN) : (window.location.href = bookingCreateRes.url);
  }
}, [bookingCreateRes?.url, deviceType, changeModalStatus]);
```